spec.emu

```diff
@@ -331,6 +359,21 @@ contributors: Google, Ecma International
         The value of the [[RelevantExtensionKeys]] internal slot is &laquo;  &raquo;.
       </p>
 
+      <p>
+        The value of the [[LocaleData]] internal slot is implementation defined within the constraints described in <emu-xref href="#sec-internal-slots"></emu-xref> and the following additional constraints:
+      </p>
+
+      <ul>
+        <li>[[LocaleData]].[[&lt;_locale_&gt;]] must have a types field for all locale values _locale_.  The value of this field must be a Record, which must have fields with the names of the six display name types: `"language"`, `"region"`, `"script"`, `"currency"`, `"dateTime"`, and `"dateTimeField"`.</li>
```

Comment:
this seems like it should be a named record type, not just a generic "Record"?

Reply:
could you point me to an example in either ECMA262 or ECMA402 that I can follow?

Reply:
https://tc39.es/ecma262/#sec-chosen-value-records, https://tc39.es/ecma262/#script-record, etc - we do indeed have some generic records, but it seems much more readable to have a linkable section with a record type instead of effectively passing around a bunch of objects.

Reply:
The pattern Frank followed here is pretty much consistent with how other ECMA-402 `[[LocaleData]]` objects are handled:

- [Intl.NumberFormat](https://www.ecma-international.org/ecma-402/6.0/index.html#sec-intl.numberformat-internal-slots)
- [Intl.DateTimeFormat](https://www.ecma-international.org/ecma-402/6.0/index.html#sec-properties-of-intl-datetimeformat-instances)

It's messy, but consistent.  If we clean it up, I'd rather do it across the spec.  I'm hoping to partially address this in tc39/ecma402#210

Reply:
so I assume  this comment is addressed. Open another Issue if we still need to do that.